### PR TITLE
LOOP-1163: Switch to using new alert facility

### PR DESCRIPTION
**Pull request by rickpasetto**
Switch DexcomCGMKit to use the new alert facility
[LOOP-1163](https://tidepool.atlassian.net/browse/LOOP-1163)

### CODE REVIEW

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift [hunk 1]
```diff
@@ -534,48 +532,45 @@ public class DexcomAlertConfigurationManager {
...
         if let signalLossConfig = mutableAlertConfiguration(forType: DexcomAlertType.signalLoss),
             signalLossConfig.enabled {
             // the signal loss alert fires after .thresholdValue minutes of no transmitter communication
             let triggerDelayMinutes = signalLossConfig.threshold.doubleValue(for: HKUnit.minute())
-            var trigger: UNTimeIntervalNotificationTrigger? = nil
-            if triggerDelayMinutes > 0 {
-                trigger = UNTimeIntervalNotificationTrigger(timeInterval: TimeInterval(minutes: triggerDelayMinutes), repeats: true)
-            }
-            scheduleNotification(forAlert: signalLossConfig, withTrigger: trigger)
+            let trigger = triggerDelayMinutes > 0 ? DeviceAlert.Trigger.repeating(repeatInterval: triggerDelayMinutes * 60) : .immediate
```
Comment (nhamming) [inline DexcomCGMKit/DexcomAlertConfigurationManager.swift:563]: [nit] if `repeatInterval` takes a `TimeInterval` consider using those semantics. If it doesn't take a `TimeInterval`, curious why not?

Comment (nhamming) [reply DexcomCGMKit/DexcomAlertConfigurationManager.swift:563]: :) this is to void such magic numbers as `60`

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift [hunk 2]
```diff
@@ -661,3 +638,21 @@ extension DexcomAlertConfigurationManager: AlertConditionChangedDelegate {
         }
     }
 }
+
+extension AlertConfiguration {
+    public func asUserAlert(with trigger: DeviceAlert.Trigger) -> DeviceAlert {
+        let userAlertInAppContent = type.hasInApp ? DeviceAlert.Content(title: type.localizedInAppTitle,
+                                                                        body: type.localizedInAppMessage,
+                                                                        acknowledgeActionButtonLabel: type.localizedDismissAction,
+                                                                        isCritical: alwaysSoundEnabled) : nil
```
Comment (nhamming) [inline DexcomCGMKit/DexcomAlertConfigurationManager.swift:647]: Unless I've missed the discussion (which is totally possible) some alerts will allows use critical alerts even when `alwaysSoundEnabled` is `false`.

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift [hunk 3]
```diff
@@ -661,3 +638,21 @@ extension DexcomAlertConfigurationManager: AlertConditionChangedDelegate {
...
+        let userAlertNotificationContent = type.hasUserNotification ? DeviceAlert.Content(title: type.localizedNotificationTitle,
+                                                                                          body: type.localizedNotificationMessage,
+                                                                                          acknowledgeActionButtonLabel: type.localizedDismissAction,
+                                                                                          isCritical: alwaysSoundEnabled) : nil
```
Comment (nhamming) [inline DexcomCGMKit/DexcomAlertConfigurationManager.swift:651]: Unless I've missed the discussion (which is totally possible) some alerts will allows use critical alerts even when alwaysSoundEnabled is false.

File: Models/DexcomAlertConfiguration.swift [hunk 4]
```diff
@@ -469,6 +517,15 @@ public enum DexcomAlertType: Int, Codable, CaseIterable {
             return nil
         }
     }
+    
+    public var localizedDismissAction: String {
+        switch self {
+        case .transmitterFound:
+            return LocalizedString("Start Session", comment: "Alert acknowledgement for Start Session button")
+        default:
```
Comment (nhamming) [inline Models/DexcomAlertConfiguration.swift:525]: Does this display only for InApp? If this displays in the UserNotification, won't this be confusing?

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift (hunk 2 above)
Comment (rickpasetto) [reply DexcomCGMKit/DexcomAlertConfigurationManager.swift:647]: ooh, good catch!  I'll fix it!

File: Models/DexcomAlertConfiguration.swift (hunk 4 above)
Comment (rickpasetto) [reply Models/DexcomAlertConfiguration.swift:525]: According to the spreadsheet, yes.  Note that there still needs to be a complete pass over all of this copy (title, body, and actions)...

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift (hunk 2 above)
Comment (rickpasetto) [reply DexcomCGMKit/DexcomAlertConfigurationManager.swift:647]: Note this _is_ using the value of `AlertConfiguration.alwaysSoundEnabled`.  Sorry, what am I missing?

File: DexcomCGMKit/DexcomAlertConfigurationManager.swift (hunk 3 above)
Comment (rickpasetto) [reply DexcomCGMKit/DexcomAlertConfigurationManager.swift:651]: see above

File: Models/DexcomAlertConfiguration.swift (hunk 4 above)
Comment (nhamming) [reply Models/DexcomAlertConfiguration.swift:525]: ok

**Pull request merged**